File: Count and Say.py
# Leetcode Problem 38: Count and Say
# PYTHON CODE
class Solution:
    def countAndSay(self, n: int) -> str:
        result = "1"

        for _ in range(n - 1):
            current = []
            count = 1

            for i in range(1, len(result)):
                if result[i] == result[i - 1]:
                    count += 1
                else:
                    current.append(str(count))
                    current.append(result[i - 1])
                    count = 1

            current.append(str(count))
            current.append(result[-1])

            result = "".join(current)

        return result

File: Count and Say.java
// Leetcode Problem 38: Count and Say
// JAVA CODE
class Solution {
    public String countAndSay(int n) {
        String result = "1";

        for (int i = 1; i < n; i++) {
            StringBuilder current = new StringBuilder();
            int count = 1;

            for (int j = 1; j < result.length(); j++) {
                if (result.charAt(j) == result.charAt(j - 1)) {
                    count++;
                } else {
                    current.append(count);
                    current.append(result.charAt(j - 1));
                    count = 1;
                }
            }

            current.append(count);
            current.append(result.charAt(result.length() - 1));

            result = current.toString();
        }

        return result;
    }
}